src/coinbase/transaction.ts

```diff
@@ -0,0 +1,135 @@
+import { ethers } from "ethers";
+import { Transaction as TransactionModel } from "../client/api";
+import { TransactionStatus } from "./types";
+import { parseUnsignedPayload } from "./utils";
+
+/**
+ * A representation of an onchain Transaction.
+ */
+export class Transaction {
+  private model: TransactionModel;
+  private raw?: ethers.Transaction;
+
+  /**
+   * Transactions should be constructed via higher level abstractions like Trade or Transfer.
+   *
+   * @class
+   * @param model - The underlying Transaction object.
+   */
+  constructor(model: TransactionModel) {
+    if (!model) {
+      throw new Error("Invalid model type");
```

Comment:
Is it sufficient that this is typed via typescript? Or is this just guarding against a nil value?

Main reason we added that to the ruby client is for better error messaging when passing in the wrong object, because there is no typing.

Reply:
The guard in the constructor is mainly to catch null values and give better error messages, similar to what we did in the Ruby client. While TypeScript handles type checking, having this check is still useful for making the code more robust and the errors clearer. If someone uses the SDK in plain JavaScript, this kind of check is even more important since JavaScript doesn't have built-in type checking.